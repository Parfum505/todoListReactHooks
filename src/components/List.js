import React from "react";
import ListItem from "./ListItem";

const List = ({ list, setCompleted, deleteTask }) => {
  return (
    <div className="row justify-content-center">
      <ul
        id="todoList"
        className="list-group col-11 col-sm-10 col-md-7 col-lg-5"
      >
        {list &&
          list.map((task) => (
            <ListItem
              key={task.id}
              task={task}
              deleteTask={deleteTask}
              setCompleted={setCompleted}
            />
          ))}
      </ul>
    </div>
  );
};

export default List;
