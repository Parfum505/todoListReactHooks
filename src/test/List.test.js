import React from "react";
import { render } from "@testing-library/react";
import List from "../components/List";

describe("List component", () => {
  const mockListItems = [
    { id: 1, item: "to learn Jest", completed: false },
    { id: 2, item: "to commit changes", completed: true },
  ];
  test("matches snapshot", () => {
    const { asFragment } = render(<List list={mockListItems} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("render listItems", () => {
    const { queryAllByTestId } = render(<List list={mockListItems} />);
    const listItems = queryAllByTestId("listItem");
    expect(listItems.length).toBe(2);
    const mockTodos = mockListItems.map((todo) => todo.item);
    const todos = listItems.map(
      (li) => li.querySelector(".form-control").value
    );
    expect(mockTodos).toEqual(todos);
  });
  test("render emty listItems", () => {
    const emptyList = [];
    const { queryAllByTestId } = render(<List list={emptyList} />);
    expect(queryAllByTestId("listItem").length).toBe(0);
  });
});
