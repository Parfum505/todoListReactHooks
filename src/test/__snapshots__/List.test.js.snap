// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center"
  >
    <ul
      class="list-group col-11 col-sm-10 col-md-7 col-lg-5"
      id="todoList"
    >
      <li
        class="input-group mb-3"
        data-testid="listItem"
        id="1"
      >
        <div
          class="custom-control custom-checkbox"
        >
          <input
            class="custom-control-input"
            data-testid="checkbox"
            id="customCheck_1"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="customCheck_1"
          >
             
          </label>
        </div>
        <input
          class="form-control "
          data-testid="taskInput"
          readonly=""
          type="text"
          value="to learn Jest"
        />
        <div
          class="input-group-append"
        >
          <button
            class="btn btn-danger input-group-text"
            data-testid="deleteBtn"
          >
            X
          </button>
        </div>
      </li>
      <li
        class="input-group mb-3"
        data-testid="listItem"
        id="2"
      >
        <div
          class="custom-control custom-checkbox"
        >
          <input
            checked=""
            class="custom-control-input"
            data-testid="checkbox"
            id="customCheck_2"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="customCheck_2"
          >
             
          </label>
        </div>
        <input
          class="form-control completed"
          data-testid="taskInput"
          readonly=""
          type="text"
          value="to commit changes"
        />
        <div
          class="input-group-append"
        >
          <button
            class="btn btn-danger input-group-text"
            data-testid="deleteBtn"
          >
            X
          </button>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
